source/tests/system/nirfmxinstr_driver_api_tests.cpp

```diff
@@ -165,6 +167,48 @@ TEST_F(NiRFmxInstrDriverApiTests, SetAndGetFrequencyReferenceSource_ReturnsFrequ
   EXPECT_EQ(response.attr_val(), "OnboardClock");
 }
 
+TEST_F(NiRFmxInstrDriverApiTests, CallCfgMethods_Succeeds)
+{
+  const auto session = init_session(stub(), PXI_5663E);
+
+  EXPECT_SUCCESS(session, client::cfg_frequency_reference(stub(), session, "", FrequencyReferenceSource::FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK, 10e6));
+  EXPECT_SUCCESS(session, client::cfg_mechanical_attenuation(stub(), session, "", MechanicalAttenuationAuto::MECHANICAL_ATTENUATION_AUTO_TRUE, 10.0));
+}
+
+TEST_F(NiRFmxInstrDriverApiTests, CallCheckMethods_SucceedsWithReasonableResponseValues)
+{
+  const auto session = init_session(stub(), PXI_5663E);
+
+  const auto status_response = EXPECT_SUCCESS(session, client::check_acquisition_status(stub(), session));
+  const auto list_exists_response = EXPECT_SUCCESS(session, client::check_if_list_exists(stub(), session, "NOTALIST"));
+  const auto self_cal_response = EXPECT_SUCCESS(session, client::is_self_calibrate_valid(stub(), session, ""));
+
+  EXPECT_EQ(status_response.acquisition_done(), Boolean::BOOLEAN_TRUE);
+  EXPECT_EQ(list_exists_response.list_exists(), Boolean::BOOLEAN_FALSE);
+  EXPECT_EQ(list_exists_response.personality(), Personality::PERSONALITY_NONE);
+  EXPECT_EQ(self_cal_response.self_calibrate_valid(), Boolean::BOOLEAN_TRUE);
+  EXPECT_THAT(self_cal_response.valid_steps_array(), ElementsAreArray(std::vector<SelfCalStep>{SelfCalStep::SELF_CAL_STEP_DIGITIZER_SELF_CAL}));
+}
+
+// Use a mix of RFmxSpecAn and RFmxInstr APIs in an RFmxInstr session.
+TEST_F(NiRFmxInstrDriverApiTests, SpectrumBasicWithRFmxInstr_DataLooksReasonable)
+{
+  auto specan_stub = create_stub<nirfmxspecan_grpc::NiRFmxSpecAn>();
+  auto session = init_session(stub(), PXI_5663);
+  EXPECT_SUCCESS(session, specan_client::cfg_rf(specan_stub, session, "", 1e9, 0, 0));
+  EXPECT_SUCCESS(session, client::cfg_frequency_reference(stub(), session, "", FrequencyReferenceSource::FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK, 10e6));
+  EXPECT_SUCCESS(session, specan_client::spectrum_cfg_span(specan_stub, session, "", 1e6));
+  EXPECT_SUCCESS(session, specan_client::spectrum_cfg_rbw_filter(specan_stub, session, "", true, 100e3, nirfmxspecan_grpc::SpectrumRbwFilterType::SPECTRUM_RBW_FILTER_TYPE_GAUSSIAN));
+  EXPECT_SUCCESS(session, specan_client::spectrum_cfg_averaging(specan_stub, session, "", false, 10, nirfmxspecan_grpc::SpectrumAveragingType::SPECTRUM_AVERAGING_TYPE_RMS));
+
+  auto read_response = specan_client::spectrum_read(specan_stub, session, "", 10.0);
+
+  EXPECT_SUCCESS(session, read_response);
+  int midpoint_x = 1005 / 2;
```

Comment:
Can you add a `EXPECT_EQ(1005, read_response.spectrum_size());` or equivalent above this line to make it clear 1005 is the size?